### DISCUSSION TITLE
Smoke Voxel Data

### DISCUSSION THREAD

**Discussion by kpei** [saul/demofile-net#124] [General]
Has anyone been able to decode the frame data?

**Comment by in0finite** [comment]
I don't know about voxel data format, but I was able to re-implement CS2 smoke on my own:
https://www.youtube.com/watch?v=xHzekB5gvwY

I could add a Console command which dumps all voxel positions into a json file, if needed.

**Reply by kpei** [reply]
Are you reading the voxel data from the demo or are you simulating the smoke itself based on the map meshes/objects?  

Btw amazing tool!

Edit: just read the youtube title. that would be great if you can add a console command for that to dump the positions and i can correlate it with the frame data.

**Reply by in0finite** [reply]
I added command `smoke_export_voxels_all` in v17.1. It should output enough data to correlate with other data sources. Right now, it only outputs minimap (2D) grid positions, not voxel (3D) positions.
